2017/test/LameingoAutonomous.java

```diff
@@ -136,8 +161,33 @@ public void init()
     @Override
     public void start()
     {
-        nearBeaconTask = new DeadReckonTask(this, approachNearBeacon, lightCriteria);
-        addTask(nearBeaconTask);
+        this.addTask(new DeadReckonTask(this, approachNearBeacon, backLightCriteria) {
+            @Override
+            public void handleEvent(RobotEvent e) {
+                if (e instanceof DeadReckonTask.DeadReckonEvent) {
+                    DeadReckonTask.DeadReckonEvent drEvent = (DeadReckonTask.DeadReckonEvent) e;
+
+                    if (drEvent.kind == DeadReckonTask.EventKind.SENSOR_SATISFIED) {
+                        doTurnOnLine();
+
+                    }
+                }
+            }
+
+            private void doTurnOnLine() {
+                robot.addTask(new DeadReckonTask(robot, lineDetectTurnPath, lightCriteria) {
+                    @Override
+                    public void handleEvent(RobotEvent e)
+                    {
+                        DeadReckonEvent drEvent = (DeadReckonEvent) e;
+
+                        if (drEvent.kind == EventKind.SENSOR_SATISFIED) {
+                            robot.addTask(new DeadReckonTask(robot, beaconAlignTurn, backLightCriteria));
+                        }
+                    }
+                });
+            }
+        });
```

Comment:
Thanks for doing this Elizabeth.  

This is almost what I was referring to, but you've nested doTurnOnLine inside the anonymous class declaration which wasn't precisely where I was going, although it's half way there.  Let's talk more on Saturday morning about how to structure this.

But good work.  I appreciate your initiative here.